File: bench/elixir/reduce.exs
n = String.to_integer(System.get_env("BENCH_N") || "10000000")
# higher-order fold: + applied per element (Enum.reduce with a passed fn, not Enum.sum).
IO.puts(Enum.reduce(0..(n - 1), 0, fn x, acc -> acc + x end))

File: bench/ruby/reduce.rb
n = (ENV["BENCH_N"] || "5000000").to_i

# higher-order fold: + applied per element via a block. NOT (0...n).sum, which
# Ruby evaluates with the Gauss closed form (O(1)) and so does no folding work.
puts (0...n).reduce(0) { |a, b| a + b }
